src/Hl7.Fhir.Base/ElementModel/ScopedNode.cs

```diff
@@ -87,6 +88,19 @@ private ScopedNode(ScopedNode parentNode, ScopedNode? parentResource, ITypedElem
         /// <inheritdoc/>
         public string Name => Current.Name;
 
+        /// <summary>
+        /// Will be replaced by a different implementation in the future.
+        /// </summary>
+        public NodeType Type => this switch
+        {
+            { AtResource: true } when Current.Children("contained").Any() => NodeType.DomainResource | NodeType.Resource,
```

Comment:
I was aware that it wasn't very solid, but for now I only use it to determine whether I should try to resolve in its contained resources, so no harm done. "HasContained" would be a better name for this since that is essentially what it is used for now, but it would violate the concept of the NodeType enum

Reply:
But if the rule is that we should not go past a DomainResource boundary, how can we make sure we don't, even if the DomainResource has no contained resources? Or is that not relevant?